### PR TITLE
feat: (Core) introduce Quick View component

### PR DESCRIPTION
**Pull request by olshansky**
Blocked by https://github.com/SAP/fundamental-styles/pull/1898

#### Please provide a link to the associated issue.
Closes: https://github.com/SAP/fundamental-ngx/issues/4071
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples

### CODE REVIEW

File: libs/core/src/lib/quick-view/quick-view-group-item-label/quick-view-group-item-label.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(QuickViewGroupItemLabelComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
```
Comment (sKudum) [inline libs/core/src/lib/quick-view/quick-view-group-item-label/quick-view-group-item-label.component.spec.ts:22]: Having test scripts will help.

File: libs/core/src/lib/quick-view/quick-view-group-item/quick-view-group-item.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(QuickViewGroupItemComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
```
Comment (sKudum) [inline libs/core/src/lib/quick-view/quick-view-group-item/quick-view-group-item.component.spec.ts:22]: Having test script will help

File: libs/core/src/lib/quick-view/quick-view-group-title/quick-view-group-title.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(QuickViewGroupTitleComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
```
Comment (sKudum) [inline libs/core/src/lib/quick-view/quick-view-group-title/quick-view-group-title.component.spec.ts:23]: Having test script will help

File: libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(QuickViewGroupComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
```
Comment (sKudum) [inline libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts:23]: Is there any reason why we don't have any specs not having test scripts?

File: libs/core/src/lib/quick-view/quick-view/quick-view.component.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    selector: 'fd-quick-view',
+    templateUrl: './quick-view.component.html',
+    styleUrls: ['./quick-view.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class QuickViewComponent {
+    /** Id of the quick view element. */
+    @Input()
+    @HostBinding('attr.id')
```
Comment (sKudum) [inline libs/core/src/lib/quick-view/quick-view/quick-view.component.ts:15]: Having @input and @hostbinding is not suggested approach.

**Review by sKudum — CHANGES_REQUESTED** [review]
Few comments from my side rest looks good.

1).If we can have what and how to use quick view component (features) will help the application developer to understand it easily.

2). *.component.spec.ts needs to have test scripts.

3). Request you to update the checklist while raising a PR (It is not updated in this PR). Its mandatory.

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
I added one comment, probably component from styles will be completely refactored, so I think it's worth to wait for styles.

File: libs/core/src/lib/quick-view/quick-view/quick-view.component.ts (hunk 5 above)
Comment (olshansky) [reply libs/core/src/lib/quick-view/quick-view/quick-view.component.ts]: @sKudum fixed

File: apps/docs/src/app/core/component-docs/quick-view/quick-view-docs.component.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        title: 'Company',
+        items: [{
+            label: 'Name',
+            value: 'Company A'
+        }, {
+            label: 'Address',
+            value: '718 Main Street, Anytown, SD 57401, USA'
+        }]
+    }]
+};
```
[the remark this answers is not in the document]
Comment (olshansky) [reply apps/docs/src/app/core/component-docs/quick-view/quick-view-docs.component.ts]: @JKMarkowski yeah, you are right, fixed

File: libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts (hunk 4 above)
Comment (olshansky) [reply libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts:23]: @sKudum Could you please guide me, which tests are you expected here? 
No one of the components has no methods or "dynamic behavior", so unit tests are not applicable for it as I see it

Comment (sKudum) [reply libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts:23]: Was wondering, testing a dummy component with template ready will help to test the component. 
such as checking whether expected tag is present and it has the value or id or class name present, is it supporting accessibility  properties. etc

Comment (olshansky) [reply libs/core/src/lib/quick-view/quick-view-group/quick-view-group.component.spec.ts:23]: @sKudum I've added tests in those places where items have binding/directive/attributes which made a sort of "dynamic behavior" 🙂

**Review by sKudum — APPROVED** [review]

**Pull request closed without merging**